DefaultQutritMixed measurement error (#5842)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

* `QuantumScript.hash` is now cached, leading to performance improvements.
  [(#5919)](https://github.com/PennyLaneAI/pennylane/pull/5919)

> * `qml.DefaultQutritMixed` readout error has been added using parameters readout_relaxation_probs and 
>   readout_misclassification_probs. These parameters add a `qml.QutritAmplitudeDamping`  and a `qml.TritFlip` channel, respectively,

Author: ```suggestion
* `~.DefaultQutritMixed` readout error has been added using parameters `readout_relaxation_probs` and 
  `readout_misclassification_probs` on the `default.qutrit.mixed` device. These parameters add a `~.QutritAmplitudeDamping`  and a `~.TritFlip` channel, respectively,
```

>   readout_misclassification_probs. These parameters add a `qml.QutritAmplitudeDamping`  and a `qml.TritFlip` channel, respectively,

Reviewer: > question: will we have a dedicated section for noisy qutrit simulation in the next release notes?

We're putting together the release notes and blog post now. For the release notes, we've kept the mention of [`default.qutrit.mixed` additions](https://xanaduai-pennylane--5928.com.readthedocs.build/en/5928/development/release_notes.html) fairly lightweight as there is a lot going on in this release. So yes I think potentially having a more expanded description in 0.38 should work :+1:

However, note that the `master` branch (which is the base of this PR) is actually aimed for 0.38. We've branched-off the 0.38 release into https://github.com/PennyLaneAI/pennylane/tree/v0.37.0-rc0. So it's ok to make any changes in this PR.

CC @isaacdevlugt

* `QuantumScript.hash` is now cached, leading to performance improvements.
  [(#5919)](https://github.com/PennyLaneAI/pennylane/pull/5919)

* `~.DefaultQutritMixed` readout error has been added using parameters `readout_relaxation_probs` and 
  `readout_misclassification_probs` on the `default.qutrit.mixed` device. These parameters add a `~.QutritAmplitudeDamping`  and a `~.TritFlip` channel, respectively,
  after measurement diagonalization. The amplitude damping error represents the potential for
  relaxation to occur during longer measurements. The trit flip error represents misclassification during readout.

>   [(#5842)](https://github.com/PennyLaneAI/pennylane/pull/5842)

Reviewer 2: Don't forget to add your name to the changelog :)

* Observable validation for `default.qubit` is now based on execution mode (analytic vs. finite shots) and measurement type (sample measurement vs. state measurement).
  [(#5890)](https://github.com/PennyLaneAI/pennylane/pull/5890)

<h4>Community contributions 🥳</h4>

> * `~.DefaultQutritMixed` readout error has been added using parameters `readout_relaxation_probs` and 

Reviewer: ```suggestion
* `DefaultQutritMixed` readout error has been added using parameters `readout_relaxation_probs` and 
```
Note that we skip these relative links because it [doesn't work in the docs](https://xanaduai-pennylane--5842.com.readthedocs.build/en/5842/development/release_notes.html), or the plain markdown version.

  to be simulated on the `default.qutrit.mixed` device.
  [(#5784)](https://github.com/PennyLaneAI/pennylane/pull/5784)

> * `qml.DefaultQutritMixed` readout error has been added using parameters damping_measurement_gammas and 

Author: Yes, `damping_measurement_gammas` is supposed to be the probabilities of relaxation.
`trit_flip_measurement_probs` is supposed to be the probabilities of misclassification flipping.

> * `qml.DefaultQutritMixed` readout error has been added using parameters damping_measurement_gammas and 

Author: I was thinking about this and I have the following replacement ideas

`damping_measurement_gammas`   ->  `readout_relaxations`
`trit_flip_measurement_probs`            ->  `readout_misclassifications`

What do you think?
